### PR TITLE
feat: add bitcoincash

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Add bitcoin cash support

example tx sent on the platform: https://blockchair.com/bitcoin-cash/transaction/21bfba7e88bc4d705c0b717c8755f5f4a595fadd96c8ea9adbd6980bc6575859

tested with sending to cashaddr format address both with and without the `bitcoincash:` prefix and to a legacy format address successfully.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

Low

## Testing

- Validate send, receive, account balance, and transaction history for bitcoincash
- Ensure all accounts load on both native wallet and keepkey

## Screenshots (if applicable)
image
image
image
image
image
image

### CODE REVIEW

File: src/components/Modals/FiatRamps/views/Manager.tsx [hunk 1]
```diff
@@ -70,6 +71,7 @@ const ManagerRouter: React.FC<ManagerRouterProps> = ({ fiatRampProvider }) => {
   const chainAdapterManager = getChainAdapterManager()
   const ethereumChainAdapter = chainAdapterManager.get(KnownChainIds.EthereumMainnet)
   const bitcoinChainAdapter = chainAdapterManager.get(KnownChainIds.BitcoinMainnet)
+  const bitcoincashChainAdapter = chainAdapterManager.get(KnownChainIds.BitcoinCashMainnet)
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/views/Manager.tsx:74]: casing if we're at it

File: src/context/PluginProvider/PluginProvider.tsx [hunk 2]
```diff
@@ -119,6 +119,7 @@ export const PluginProvider = ({ children }: PluginProviderProps): JSX.Element =
       if (!featureFlags.Osmosis && chainId === KnownChainIds.OsmosisMainnet) return false
       if (!featureFlags.Avalanche && chainId === KnownChainIds.AvalancheMainnet) return false
       if (!featureFlags.Litecoin && chainId === KnownChainIds.LitecoinMainnet) return false
+      if (!featureFlags.Bitcoincash && chainId === KnownChainIds.BitcoinCashMainnet) return false
```
Comment (0xdef1cafe) [inline src/context/PluginProvider/PluginProvider.tsx:122]: casing

File: src/plugins/bitcoincash/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { KnownChainIds } from '@shapeshiftoss/types'
+import * as unchained from '@shapeshiftoss/unchained-client'
+import { getConfig } from 'config'
+import { type Plugins } from 'plugins/types'
+
+// eslint-disable-next-line import/no-default-export
+export default function register(): Plugins {
+  return [
+    [
+      'bitcoincashChainAdapter',
```
Comment (0xdef1cafe) [inline src/plugins/bitcoincash/index.tsx:12]: casing

File: src/state/slices/assetsSlice/assetsSlice.ts [hunk 4]
```diff
@@ -53,12 +59,17 @@ export const assetApi = createApi({
...
-        const { Avalanche, Osmosis, Litecoin } = selectFeatureFlags(getState() as ReduxState)
+        const { Avalanche, Bitcoincash, Osmosis, Litecoin } = selectFeatureFlags(
+          getState() as ReduxState,
+        )
 
         const service = await getAssetService()
         const assets = Object.entries(service?.getAll() ?? {}).reduce<AssetsById>(
           (prev, [assetId, asset]) => {
             if (!Avalanche && asset.chainId === avalancheChainId) return prev
+            if (!Bitcoincash && asset.chainId === bchChainId) {
```
Comment (0xdef1cafe) [inline src/state/slices/assetsSlice/assetsSlice.ts:70]: *eye twitch* - one line

Comment (kaladinlight) [reply src/state/slices/assetsSlice/assetsSlice.ts:70]: haha don't know how that one happened

File: src/hooks/useTxDetails/useTxDetails.ts [hunk 5]
```diff
@@ -140,13 +141,15 @@ export const useTxDetails = (txId: string, activeAsset?: Asset): TxDetails => {
   const [ensTo, setEnsTo] = useState<string>()
 
   useEffect(() => {
+    if (tx.chainId !== ethChainId) return
```
Comment (gomesalexandre) [inline src/hooks/useTxDetails/useTxDetails.ts:144]: ❤️

**Review by gomesalexandre — COMMENTED** [review]
Logic looks sound and similar to the one from other UTXOs

Tested:

### Empty Tx history

image

More testing to be done after receiving test funds

### Accounts

image

**Review by elmutt — DISMISSED** [review]

**Review by gomesalexandre — DISMISSED** [review]
Tested again locally with testing funds on BCH account:

- Balance/charts properly accounts for the BCH value with the flag on/disregards it with the flag off - same with DeFi net worth value
- Dashboard assets shows BCH
- Asset page for BCH is working with balance/price charts - 1M is rugged but this is not related to this PR - most likely that bug we have with positive UTC offsets?
- Tx history is working on accounts/assets/dashboard/Tx history pages

🍏

**Review by gomesalexandre — APPROVED** [review]
Approved as previously tested, and the lib PR just brought casing changes which CI doesn't complain about

**Pull request merged**